### ISSUE TITLE
Fixture factories in Plugins & Packages

### ISSUE THREAD

**Issue by pabloelcolombiano** [cakephp/docs#7217]
**Issue Description**

This is a discussion.

Since over two years, and particularly since the last Cakefest 2021, we are receiving ever more cheering for the [CakePHP Fixture Factories plugin](https://github.com/vierge-noire/cakephp-fixture-factories).

The idea of generating test fixtures with factories is not new, and it is regarded as a good practice for testing.

To my knowledge, only Laravel and Symfony have similar features in PHP.
[In Laravel](https://laravel.com/docs/8.x/database-testing)
[In Symfony](https://symfony.com/bundles/ZenstruckFoundryBundle/current/index.html)
[Factory Muffin](https://github.com/thephpleague/factory-muffin)

By making use of the powerful [CakePHP ORM](https://github.com/cakephp/orm), the fixture factories have arguably some advantages over the solutions above. One of them is the possibility to generate infinitely deep associated data in an intuitive manner, with the well none dotted notation, such as ``AuthorFactory::make()->with('Address.City.Country')``.

But the best out of the ORM, is that factories can be used without Table objects. We have implemented a solution where the associations can be defined within the factories, in an ``initialize()`` method, making the factories look a bit like a behavior, from the syntactic point of view. In other words, CakePHP fixture factories are stand-alone and can be used to inject test data in any PHP projects, regardless of the framework used. 

Long story short, the plugin eases significantly the software development process, and we are trying to give the plugin a larger audience both within the CakePHP developers community, and generally for the whole PHP community, thus indirectly spreading the CakePHP framework. 

My suggestion would be to create a clean documentation of the fixture factories in the cookbook under the `Plugins & Packages` section. I would then point to this documentation when presenting the plugin to the PHP community.

I could provide both English and French translations.

This is open to discussion, before undertaking this task, I'd be happy to get your comments and feel the traction on that topic.

**Comment by othercorey** [comment]
Are you able to build github pages for plugin docs? We could point to that.

https://pages.github.com/

**Comment by pabloelcolombiano** [comment]
@othercorey indeed, I was about to do a clean doc with GitHub pages. But then wondered if the fixture factories should be hosted on the Cookbook, for the reasons mentioned above. For the moment, they are referenced [here](https://book.cakephp.org/4/en/development/testing.html#fixture-factories), pointing to the repository.

**Comment by othercorey** [comment]
I think separate gh pages docs would be easier to maintain. We won't have to update the main cookbook with external plugin changes.

**Comment by ADmad** [comment]
Maintaining docs is not a trivial task and the CakePHP team can't take up the additional work of managing docs for community plugins.

You need to manage them yourself. For e.g. the Crud plugin has it's own [docs](https://crud.readthedocs.io/en/latest/).

**Comment by pabloelcolombiano** [comment]
I get your concern @othercorey and @ADmad. I've reworked the documentation on the repository directly. It is much easier to have the docs close to the source code, so I'll keep it like this for the moment. But thanks for the link to the Crud plugin's doc, that will be the next step.
